### PR TITLE
Retain executor cache counts

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

When an executor cache entry is re-added its counts are cleared, wiping out its history even though it might be used frequently

#### Summary of Changes

Retain those counts

Fixes #

### CODE REVIEW

**Comment by jackcmay** [comment]
@t-nelson @jstarry This is already probably part of the eviction rework but in case not we should probably retain these counts on re-add

**Comment by jstarry** [comment]
Looks like a good incremental improvement to me, just needs tests

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -404,29 +404,35 @@ impl CachedExecutors {
...
-            let mut least = u64::MAX;
-            let default_key = Pubkey::default();
-            let mut least_key = &default_key;
-
-            for (key, entry) in self.executors.iter() {
-                let count = entry.prev_epoch_count + entry.epoch_count.load(Relaxed);
-                if count < least {
-                    least = count;
-                    least_key = key;
+        let entry = if let Some(entry) = self.executors.get(pubkey) {
```
Comment (t-nelson) [inline runtime/src/bank.rs:407]: If we `remove()` instead of `get()` here, the entry can be reused and avoid the atomic ops

**Review by t-nelson — APPROVED** [review]
lgtm!  Just one suggestion for you to take or leave

**Review by jstarry — APPROVED** [review]

**Pull request merged**